commands/essentials/init-bot.js

```diff
@@ -1,332 +1,343 @@
+            sendMsgToChannel(channel, userId, 'Report command is not enabled.', 8);
+            adminConsole.addField('Report Feature', 'IS NOT ENABLED!');
+        }
+
+
+        // ask if the user wants to use the experimental !ask command
+        var isAsk;
+        try {
+            isAsk = await SpecialPrompt.boolean({ prompt: 'Do you want to let users use the experimental !ask command?', channel, userId });
+        } catch {
+            isAsk = false;
+        }
+        if (isAsk) {
+            botGuild.setUpAsk(this.client);
+            sendMsgToChannel(channel, userId, 'The ask command is now available to the server users.', 8);
+            adminConsole.addField('Experimental Ask Command', 'IS ENABLED!');
+        } else {
+            sendMsgToChannel(channel, userId, 'Ask command is not enabled.', 8);
+            adminConsole.addField('Experimental Ask Command', 'IS NOT ENABLED!');
+        }
+
+        await botGuild.save();
+        botGuild.setCommandStatus(this.client);
+
+        sendMsgToChannel(channel, userId, 'The bot is set and ready to hack!', 8);
+    }
+
+    /**
+     * @typedef TypeInfo
+     * @property {String} type
+     * @property {String} roleId
+     */
+
+    /**
+     * Prompts the user for a verification type and if they want to add more. Will call itself if true 
+     * for a recursive call.
+     * @param {TextChannel} channel 
+     * @param {String} userId 
+     * @returns {Promise<TypeInfo[]>}
+     * @async
+     */
+    async getVerificationTypes(channel, userId, guestRole, memberRole) {
+        let {type, role} = await this.getValidVerificationTypes(channel, userId, guestRole, memberRole);
+
+        if (await SpecialPrompt.boolean({ prompt: 'Would you like to add another verification option?', channel, userId })) {
+            return (await this.getVerificationTypes(channel, userId, guestRole, memberRole)).concat([{
+                type: type,
+                roleId: role.id,
+            }]);
+        } else {
+            return [{
+                type: type,
+                roleId: role.id,
+            }];
+        }
+    }
+
+    async getValidVerificationTypes(channel, userId, guestRole, memberRole) {
+        let typeMsg = await MessagePrompt.prompt({
+            prompt: `Please tell me the type and mention the role for a verification option. 
+            For example: hacker @hacker . Make sure you add nothing more to the message!`, channel, userId
+        });
+        let type = typeMsg.content.replace(/<(@&?|#)[a-z0-9]*>/, ''); // clean out any snowflakes
+        type = type.toLowerCase().trim();
+        let role = typeMsg.mentions.roles.first();
+
+        if (role.id === guestRole.id || role.id === memberRole.id) {
+            sendMsgToChannel(channel, userId, 'Guest and member roles cannot be used for verification. ' +
+                'Please try again.', 30);
+            await this.getValidVerificationTypes(channel, userId, guestRole, memberRole);

```

Comment:
The return of this call is never used? So this won't work?

Reply:
Good refactor tho!

Reply:
oh good point, it should probably be like `return await this.getValidVerificationTypes(channel, userId, guestRole, memberRole);`

Reply:
added the return!